build/py.sh

```diff
@@ -25,16 +25,24 @@ export PYTHONPATH='.:vendor/'
 
 ubuntu-deps() {
   # python-dev: for all the extension modules
-  #   TODO: upgrade Ubuntu and change to python2-dev
   # gawk: used by spec-runner.sh for the special match() function.
   # time: used to collect the exit code and timing of a test
   # libreadline-dev: needed for the build/prepare.sh Python build.
   # cmake: for build/py.sh yajl-release
   set -x  # show what needs sudo
 
   # pass -y for say gitpod
+
+  ubuntu_release_year=$(lsb_release -rs | cut -d. -f1)
+
+  if $(expr $ubuntu_release_year > 20); then
```

Comment:
Please try 

```
if apt-get show python2-dev >/dev/null; then
 ...

```


So this will   prefer the python2-dev package if it exists


See this page for background

https://github.com/oilshell/oil/wiki/Feature-Detection-Is-Better-than-Version-Detection


(Also note the `expr` usage and `if` usage aren't correct here  -- you're doing a redirect with `>` , not greater than, and if tests the exit code and not the arg value)



```
$ apt-cache show python2-dev; echo $?
N: Unable to locate package python2-dev
E: No packages found
100
andy@lenny:~/git/oilshell/oil$ apt-cache show python-dev; echo $?
Package: python-dev
Architecture: amd64
```